sourcemash/frontend/static/js/views/browse_view.js

```diff
@@ -0,0 +1,66 @@
+Sourcemash.Views.BrowseView = Backbone.View.extend({
+  template: JST['browse'],
+
+  initialize: function (options) {
+    var ExtendedTypeahead = Backbone.Typeahead.extend({
+      template: JST['new-feed-form'],
+    });
+
+    this.typeahead = new ExtendedTypeahead({collection: this.collection, key: 'title'});
+    this.listenTo(this.collection, 'sync', this.render);
+  },
+
+  events: {
+    'submit #add_feed_form': 'createFeed',
+  },
+
+  createFeed: function(e){
+  	e.preventDefault()
+    this.collection.create(this.newAttributes(), {success: this.updateCollection});
+  },
+
+  updateCollection: function(newFeed) {
+  	this.$('#url').val('');
+  	newFeed.collection.fetch();
+    toast('Feed added!', 3000)
+
+    mixpanel.track("Subscribed", { "Feed Title": newFeed.get('title'),
```

Comment:
We should add a source to subscribed on feed page toggle, if it isn't there yet.